src/main/java/seedu/address/logic/commands/add/AddApplicationCommand.java

```diff
@@ -67,32 +64,18 @@ public AddApplicationCommand(Index companyIndex, Index internshipIndex, Status s
     public CommandResult execute(Model model) throws CommandException {
         requireNonNull(model);
 
-        if (!model.getTabName().equals(TabName.APPLICATION)) {
-            throw new CommandException(String.format(MESSAGE_WRONG_TAB, APPLICATION_NAME));
-        }
-
-        List<CompanyItem> lastShownCompanyList = model.getCompanyList().getFilteredItemList();
-
-        if (companyIndex.getZeroBased() >= lastShownCompanyList.size()) {
-            throw new CommandException(String.format(Messages.MESSAGE_INVALID_ITEM_DISPLAYED_INDEX, COMPANY_NAME));
-        }
-
-        CompanyItem companyItem = lastShownCompanyList.get(companyIndex.getZeroBased());
-        List<InternshipItem> internshipItemList = companyItem.getInternships();
-
-        if (internshipIndex.getZeroBased() >= internshipItemList.size()) {
-            throw new CommandException(String.format(Messages.MESSAGE_INVALID_ITEM_DISPLAYED_INDEX, INTERNSHIP_NAME));
-        }
-
-        InternshipItem internshipItem = internshipItemList.get(internshipIndex.getZeroBased());
+        InternshipItem internshipItem = InternshipAccessor.getInternship(model, companyIndex, internshipIndex);
         ApplicationItem applicationItem = new ApplicationItem(internshipItem, status, statusDate);
 
         FilterableItemList<ApplicationItem> applicationList = model.getApplicationList();
 
         if (applicationList.hasItem(applicationItem)) {
-            throw new CommandException(MESSAGE_DUPLICATE_APPLICATION);
+            throw new CommandException(String.format(MESSAGE_DUPLICATE_ITEM, APPLICATION_NAME));
         }
+
         applicationList.addItem(applicationItem);
+        model.setTabName(TabName.APPLICATION);
+
         return new CommandResult(String.format(MESSAGE_SUCCESS, applicationItem));
```

Comment:
Just tried successfully on my end for add company command, regarding the switching of tabs, changing this to `return new CommandResult(String.format(MESSAGE_SUCCESS, applicationItem), false, false, true);` will do the trick!

Reply:
I think we could just set the default value for isSwitchTab to be true then?